pkg/agent/flowexporter/connections/connections.go

```diff
@@ -122,6 +122,11 @@ func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
 	if exists {
+		// no need to update the connection when tcp state is "TIME_WAIT"
+		if existingConn.TCPState == "TIME_WAIT" {
+			cs.DeleteConnectionByKey(connKey)
+			return
+		}
```

Comment:
You should wait for the record to be sent after idleFlowTimeout for this connection, once it state becomes `TIME_WAIT`. 
Please see here:  https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/flowexporter/exporter/exporter.go#L303

Reply:
> Thanks @zyiou for working on this.
> Please evaluate if we still need `isPresent` field in `connection struct`.

There are two circumstances we need to delete the connection and flow record:
1. the connection does not exist in the conntrack table
2. the flow record has timed out and sent, and the tcp state of connection is TIME_WAIT
These two conditions may not be exactly the same, so keeping IsPresent is necessary to ensure obsolete connections are all deleted. Additionally, the delete step needs to happen in exporter because we want to make sure the record has been sent before deleting them.
I added an `IsRemoved` variable in connection to mark the 2nd case so that we can avoid adding the connection again when polling from conntrack table.